docs/source/models.rst

```diff
@@ -22,819 +22,27 @@ keypoint detection, video classification, and optical flow.
-``references/segmentation/coco_utils.py``. The classes that the pre-trained model outputs are the following,
-in order:
+Table of all available classificaiton weights
+---------------------------------------------
 
-  .. code-block:: python
+Accuracies are reported on ImageNet
 
-      ['__background__', 'aeroplane', 'bicycle', 'bird', 'boat', 'bottle', 'bus',
-       'car', 'cat', 'chair', 'cow', 'diningtable', 'dog', 'horse', 'motorbike',
-       'person', 'pottedplant', 'sheep', 'sofa', 'train', 'tvmonitor']
+.. include:: generated/classification_table.rst
 
-The accuracies of the pre-trained models evaluated on COCO val2017 are as follows
-
-================================  =============  ====================
-Network                           mean IoU       global pixelwise acc
-================================  =============  ====================
-FCN ResNet50                      60.5           91.4
-FCN ResNet101                     63.7           91.9
-DeepLabV3 ResNet50                66.4           92.4
-DeepLabV3 ResNet101               67.4           92.4
-DeepLabV3 MobileNetV3-Large       60.3           91.2
-LR-ASPP MobileNetV3-Large         57.9           91.2
-================================  =============  ====================
-
-
-Fully Convolutional Networks
-----------------------------
-
-.. autosummary::
-    :toctree: generated/
-    :template: function.rst
-
-    torchvision.models.segmentation.fcn_resnet50
-    torchvision.models.segmentation.fcn_resnet101
-
-
-DeepLabV3
----------
-
-.. autosummary::
-    :toctree: generated/
-    :template: function.rst
-
-    torchvision.models.segmentation.deeplabv3_resnet50
-    torchvision.models.segmentation.deeplabv3_resnet101
-    torchvision.models.segmentation.deeplabv3_mobilenet_v3_large
-
-
-LR-ASPP
--------
-
-.. autosummary::
-    :toctree: generated/
-    :template: function.rst
-
-    torchvision.models.segmentation.lraspp_mobilenet_v3_large
-
-.. _object_det_inst_seg_pers_keypoint_det:
 
 Object Detection, Instance Segmentation and Person Keypoint Detection
 =====================================================================
 
-The models subpackage contains definitions for the following model
-architectures for detection:
-
-- `Faster R-CNN <https://arxiv.org/abs/1506.01497>`_
-- `FCOS <https://arxiv.org/abs/1904.01355>`_
-- `Mask R-CNN <https://arxiv.org/abs/1703.06870>`_
-- `RetinaNet <https://arxiv.org/abs/1708.02002>`_
-- `SSD <https://arxiv.org/abs/1512.02325>`_
-- `SSDlite <https://arxiv.org/abs/1801.04381>`_
-
-The pre-trained models for detection, instance segmentation and
-keypoint detection are initialized with the classification models
-in torchvision.
-
-The models expect a list of ``Tensor[C, H, W]``, in the range ``0-1``.
-The models internally resize the images but the behaviour varies depending
-on the model. Check the constructor of the models for more information. The
-output format of such models is illustrated in :ref:`instance_seg_output`.
```

Comment:
I understand that your plan is to add back this info on follow up PRs. My concern about removing this info now is that we will no longer have this info committed on our main branch making it harder to recover. This is likely to be more of a problem if we consider involving the community.

Have you considered removing only the bits that you actually tackle in this PR (for example the overall accuracy table for Classification) + ResNet/VGG model builders? This way the info of anything not completed will be maintained and cleaned up on follow up PRs.

An alternative way to maintain the info is to create a feature branch for the entire documentation work and merge it at the very end on main when you are ready (similar to what we did with Multi-weight support).

Reply:
I'm in favour of starting from a clean slate here. By keeping the previous text, it's likely that we will forget removing some obsolete info. Re-writing the narrative part of docs is part of the TODOs in my original comment above, so this is definitely something that we'll need to tackle eventually.

I'm not sure I get your point regarding this making this difficult if we involve the community. From my point of view, it will make things easier since we won't have to worry about contributors editing the text in various places. But perhaps I'm missing something?

Regarding the use of a new branch, are you concerned about the live docs at [https://pytorch.org/vision/main/](https://pytorch.org/vision/main/)  missing some info?

Reply:
Thanks for the additional info. I agree that the current docs are far from good and are noisy because they provide lots of information about the models in an unstructured way. For example:
1. They explain how to use the models during inference.
2. They talk about the training data and the normalization values.
3. They provide random info on where the pre-trained models are stored, the `train()/eval()` calls, the accuracies, etc

Some of this information might be obsolete (for example all the inference info should be re-written with the presets in mind) but some is still valid and just needs to be transferred to the right place (for example the category mapping between Pascal and COCO for segmentation, info related to where the pretrained weights are stored etc). My concern for removing the latter prematurely is that for a period the live docs will be missing critical information. Correct me if I'm wrong but I understand that after merging this, we won't even have a list of the available models linking from the documentation. This is why I said earlier that it could be hard for the community to restore some of this info (they would have to dig on the previous versions of the documentation and recover the useful parts which got deleted).

One approach to mitigate the above is to only remove the info you replace. We can follow the same approach in every PR sent by the community (aka that the transfer of info on the new format is correct and that things were cleaned up correctly). We can also ensure that a deep cleaning is done at the end by reviewing the documentation as a whole (similar to what we did with the final clean up steps at #4652).

As mentioned above, another approach could be to keep a feature branch (which will certainly allow you to start from a clean slate) but the problem with that is you don't want to leave this open for too long to avoid conflicts. Also the community won't be credited for their PRs because in the end only 1 PR will land on main. 

Let me know your thoughts. Happy to have an offline chat if you prefer to speed this up.

Reply:
Thanks for the details @datumbox ! As discussed offline, I moved the new models docs into a completely separate `models_new.rst` file. This keeps the current models docs untouched, and we'll be able to make PRs against `main`. Once all is done, all we'll have to do is to rename `models_new.rst` into `models.rst` (and fix all the broken references ;) ).